payload/Library/Scripts/Install or Defer.sh

```diff
@@ -621,12 +621,11 @@ MSG_INSTALL="$(echo "$MSG_INSTALL" | /usr/bin/sed "s/%SUPPORT_CONTACT%/${SUPPORT
 MSG_INSTALL_NOW="$(echo "$MSG_INSTALL_NOW" | /usr/bin/sed "s/%SUPPORT_CONTACT%/${SUPPORT_CONTACT}/")"
 MSG_UPDATING="$(echo "$MSG_UPDATING" | /usr/bin/sed "s/%SUPPORT_CONTACT%/${SUPPORT_CONTACT}/")"
 
-# Check for updates, exit if none found, otherwise continue.
-check_for_updates
-
 # Perform first-run tasks, including calculating deadline.
 FORCE_DATE=$(/usr/bin/defaults read "$PLIST" UpdatesForcedAfter 2>"/dev/null")
 if [[ -z "$FORCE_DATE" || "$FORCE_DATE" -gt $(( $(/bin/date +%s) + MAX_DEFERRAL_TIME )) ]]; then
+    # Check for updates, exit if none found, otherwise continue.
+    check_for_updates
```

Comment:
The update list is generated from this function run, then included in subsequent user-facing alerts (overwriting the `%UPDATE_LIST%` placeholder). If `check_for_updates` is only ever run the first time the script is executed, won't this update list be missing from future script runs? The reruns of the update list check also ensure that `%UPDATE_LIST%` is kept up to date on future script runs in case some updates were run but not others.

Reply:
The rerun bit probably isn't important here since it still runs as part of later script actions, but we'd need some way to record pending updates between script runs (maybe write to a temp file on disk?), and also need a way to clean up the script once updates are run successfully (which currently checks for updates as a final validation step).

Reply:
Correct,

i have the first bit solved and I’ll upload the code sometime today.

it just outputs the update _list to the PLIST file and reads it back in when the script is executed each time

Reply:
commits added for exporting the update list into the preference file and importing the values back into the script.